### PR TITLE
N+1

### PR DESCRIPTION
**Pull request by Svartstein**
## In this pull request:

n+1
- N+1 problem when fetching all posts and their comments for a user solved.
  - Before:
image_2023-08-29_12-19-37

  - After:
image_2023-08-29_12-20-04

Integration tests made for:
- User index page:
- User show page:
- Post index page:
- Post show page:

### CODE REVIEW

**Comment by prolajumokeoni** [comment]
Hi @Team

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has Rubocop linter errors, please make sure as partners you have balanced commits 👍🏿 
 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/integration/posts_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      text: 'In martial arts, the way of the water is the way of the soft and yielding.'
+    )
+  end
+  let!(:post2) do
+    Post.create(
+      author:,
+      title: 'The way of the Fire',
+      text: 'In martial arts, the way of the fire is the way of the hard and unyielding.'
+    )
+  end
```
Comment (DuaneDave) [inline spec/integration/posts_index_spec.rb:19]: Good job, writing some of the required integrations, however, there are many more you've skipped. I recommend that you implement these specs in order to satisfy the milestone's requirement:

- Tests that asserts you can see the user's profile picture.
- Tests that asserts you can see the user's username.
- Tests that asserts you can see a post's title.
- Tests that asserts you can see some of the post's body.
- Tests that asserts you can see the first comments on a post.
- Tests that asserts you can see how many comments a post has.
- Tests that asserts you can see how many likes a post has.
- Tests that asserts you can see a section for pagination if there are more posts than fit on the view.

File: spec/integration/posts_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      post:,
+      text: 'I prefer the way of the leaf, which is the way of the ninja.'
+    )
+  end
+  let!(:comment2) do
+    Comment.create(
+      user: author,
+      post:,
+      text: 'The way of the leaf is the way of the coward.'
+    )
```
Comment (DuaneDave) [inline spec/integration/posts_show_spec.rb]: Good job, writing some of the required integrations, however, there are many more you've skipped. I recommend that you implement these specs in order to satisfy the milestone's requirement:

- Tests that asserts you can see the comment each commentor left.
- Tests that asserts you can see the username of each commentor.

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @Swarzstein and @enmanuelbayen  👋 , 

Good job so far on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
✔️ Capybara is present i gem file
✔️ Most specs were correctly done

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/posts_index_spec.rb (hunk 1 above)
Comment (VICTOR-CHUKWU) [reply spec/integration/posts_index_spec.rb:19]: - [x] Good job in adding the tests required by the previous reviewer, but you missed a  `Tests that asserts you can see how many likes a post has`. kindly fix it

File: spec/integration/posts_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                                 'ninjas are not cowards ' \
+                                 'but warriors who fight in the shadows.')
+  end
+
+  it 'displays the comment author for each comment' do
+    visit user_post_path(author, post)
+    expect(page).to have_content('Kenshiro')
+    expect(page).to have_content('Akai321')
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline spec/integration/posts_show_spec.rb]: - [x] you have some tests failing, kindly fix that
image

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Swarzstein , 
## STATUS: CHANGES REQUIRED :recycle: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by jerryowusu — APPROVED** [review]
Hi @Team

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

200w


## Highlights
Made requested changes :+1: 
All tests are passing :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**